src/viam/examples/dial/example_dial.cpp

```diff
@@ -63,7 +63,11 @@ int main() {
 
     // ensure we can create clients to the robot
     auto gc = robot->resource_by_name<GenericClient>("generic1");
-    std::cout << "got generic client named " << gc->name() << std::endl;
+    if (gc) {
```

Comment:
(flyby) add a guardrail here just in case the robot someone tries to use doesn't have a generic named `generic1`, which causes a segfault (and therefore a failure to close the robot successfully).